File: src/core/list_item.rs
use chrono::NaiveDateTime;
use uuid::Uuid;

#[derive(Debug, Clone)]
pub struct ListItem {
    pub id: Uuid,
    pub title: String,
    pub notes: String,
    pub created: NaiveDateTime,
    pub done: bool,
}

impl ListItem {
    pub fn new(title: impl Into<String>) -> Self {
        Self {
            id: Uuid::new_v4(),
            title: title.into(),
            notes: String::new(),
            created: chrono::Local::now().naive_local(),
            done: false,
        }
    }
}

File: lamp-mobile/core/model/src/main/java/com/lamp/mobile/core/model/ListItem.kt
package com.lamp.mobile.core.model

import java.time.LocalDateTime
import java.util.UUID

enum class ListKind {
    MEDIA,
    SHOPPING,
}

data class ListItem(
    val id: UUID = UUID.randomUUID(),
    val title: String,
    val notes: String = "",
    val created: LocalDateTime = LocalDateTime.now(),
    val done: Boolean = false,
    val kind: ListKind = ListKind.MEDIA,
)
